File: judo-sdk/src/main/java/com/judopay/secure3d/JsonParsingJavaScriptInterface.java
package com.judopay.secure3d;

import static android.text.TextUtils.isEmpty;

class JsonParsingJavaScriptInterface {

    private final JsonListener jsonListener;

    JsonParsingJavaScriptInterface(JsonListener jsonListener) {
        this.jsonListener = jsonListener;
    }

    @android.webkit.JavascriptInterface
    public void parseJsonFromHtml(final String content) {
        if (!isEmpty(content)) {
            try {
                String json = content.substring(content.indexOf("{"), content.lastIndexOf("}") + 1);
                jsonListener.onJsonReceived(json);
            } catch (StringIndexOutOfBoundsException ignore) {
            }
        }
    }

    interface JsonListener {
        void onJsonReceived(String json);
    }

}

File: judokit-android/src/main/java/com/judokit/android/ui/cardverification/component/JsonParsingJavaScriptInterface.kt
package com.judokit.android.ui.cardverification.component

class JsonParsingJavaScriptInterface(private val onJsonReceived: (String) -> Unit) {

    @android.webkit.JavascriptInterface
    fun parseJsonFromHtml(content: String?) {
        if (!content.isNullOrEmpty()) {
            try {
                val json = content.substring(content.indexOf("{"), content.lastIndexOf("}") + 1)
                onJsonReceived.invoke(json)
            } catch (ignore: StringIndexOutOfBoundsException) {
            }
        }
    }
}
